LayoutTests/ipc/coreipc.js

```diff
@@ -491,32 +491,56 @@ export class ArgumentSerializer {
     }
 
     static serializeKeyValuePair(innerType, argument) {
-        const splitTemplateType = ArgumentSerializer.splitTemplateType(innerType);
-        if(argument['key'] === undefined)
-            throw new SerializationError('argument of key value type is missing the key attribute');
-        if(argument['value'] === undefined)
-            throw new SerializationError('argument of key value type is missing the key attribute');
-        const result = [];
-        let argumentDefinition = {type: splitTemplateType[0], name: "key"};
-        result.push(ArgumentSerializer.serializeArgument(argumentDefinition, argument['key']));
-        argumentDefinition = {type: splitTemplateType[1], name: "value"};
-        result.push(ArgumentSerializer.serializeArgument(argumentDefinition, argument['value']));
-        return result;
+        // we handle both [key, value] and {key, value} formats
+
+        if(Array.isArray(argument) && argument.length == 2) {
+            return ArgumentSerializer.serializePair(innerType, argument);
+        }
+
+        if(argument['key'] !== undefined && argument['value'] !== undefined) {
+            const splitTemplateType = ArgumentSerializer.splitTemplateType(innerType);
+            const result = [];
+            let argumentDefinition = {type: splitTemplateType[0], name: "key"};
+            result.push(ArgumentSerializer.serializeArgument(argumentDefinition, argument['key']));
+            argumentDefinition = {type: splitTemplateType[1], name: "value"};
+            result.push(ArgumentSerializer.serializeArgument(argumentDefinition, argument['value']));
+            return result;
+        }
+
+        throw new SerializationError('argument of KeyValuePair is not an array with two elements or an object with {key, value} properties');
     }
 
     static serializeVariant(innerType, argument) {
         const variantTypes = ArgumentSerializer.splitTemplateType(innerType);
-        if(!Object.hasOwn(argument, 'variantType')) {
-            throw new SerializationError('missing field \'variantType\' when serializing variant');
+        let variantType = undefined;
+        let variantIndex = undefined;
+
+        if(Object.hasOwn(argument, 'variantType')) {
+            variantType = argument.variantType;
+            variantIndex = variantTypes.indexOf(argument.variantType);
+
+            if(variantIndex == -1) {
+                throw new SerializationError(`type '${ variantType }' is not valid for variant (${ variantTypes })`);
+            }
+        }
+
+        if(Object.hasOwn(argument, 'variantIndex')) {
+            variantIndex = argument.variantIndex;
+            variantType = variantTypes[variantIndex];
```

Comment:
Do we want to make sure `argument` does not have both? 
Or if it can have both, maybe we should verify the  computed `argument.variantIndex` matches `variantIndex` instead of just overwriting `variantIndex`?